pkg/sriov/sriov.go

```diff
@@ -176,7 +176,27 @@ func (s *sriovManager) ReleaseVF(conf *sriovtypes.NetConf, podifName string, cid
 				return fmt.Errorf("failed to parse original effective MAC address %s: %v", conf.OrigVfState.EffectiveMAC, err)
 			}
 
-			if err = s.nLink.LinkSetHardwareAddr(linkObj, hwaddr); err != nil {
+			/* Some NIC drivers (i.e. i40e/iavf) set VF MAC address asynchronously.
+			   This means that we can get an error from the setHardwareAddr, or it may just fail silently
+			   so to cover both cases we will retry the set function
+			   and also check the mac address on the VF after the configuration.
+			*/
+			err = utils.Retry(10, 100*time.Millisecond, func() error {
```

Comment:
we have two places where we set the MAC netdev mac address (here and in L104)

shall we use the same logic in both ? it seems the "fail silently" part may happen on set as well WDYT ?

that is just have a : `setVFEffectiveMAC(netdevName, MAC)` and call it with utils.Retry

Reply:
done can you please have a look?